concrete/src/Area/Layout/Preset/Provider/ActiveThemeProvider.php

```diff
@@ -1,34 +1,98 @@
 <?php
 namespace Concrete\Core\Area\Layout\Preset\Provider;
 
-use Page;
+use Concrete\Core\Page\Theme\Theme;
+use Concrete\Core\Http\Request;
+use Concrete\Core\Site\Service as SiteService;
 
 class ActiveThemeProvider implements ProviderInterface
 {
-    protected $presets = array();
+    /**
+     * The name of this provider.
+     *
+     * @var string
+     */
+    protected $name;
 
-    public function __construct()
+    /**
+     * The available layout presets.
+     *
+     * @var \Concrete\Core\Area\Layout\Preset\PresetInterface[]
+     */
+    protected $presets;
+
+    /**
+     * The theme that provides the presets.
+     *
+     * @var \Concrete\Core\Page\Theme\Theme|null
+     */
+    protected $theme;
+
+    public function __construct(Request $request, SiteService $siteService)
     {
-        $c = Page::getCurrentPage();
-        if (is_object($c)) {
+        $theme = null;
+        $c = $request->getCurrentPage();
+        if ($c && !$c->isError()) {
             $theme = $c->getCollectionThemeObject();
-            if (is_object($theme)) {
-                if ($theme instanceof ThemeProviderInterface) {
-                    $provider = new ThemeProvider($theme);
-                    $this->presets = $provider->getPresets();
-                    $this->name = $provider->getName();
-                }
+        }
+        if ($theme === null) {
+            $site = $siteService->getActiveSiteForEditing();
+            if ($site !== null) {
+                $themeID = $site->getThemeID();
+                $theme = $themeID ? Theme::getByID($themeID) : null;
             }
         }
+        if ($theme instanceof Theme) {
+            $this->theme = $theme;
+            $this->name = $theme->getThemeDisplayName();
+            if ($theme instanceof ThemeProviderInterface) {
+                $provider = new ThemeProvider($theme);
+                $this->presets = $provider->getPresets();
+            } else {
+                $this->presets = [];
+            }
+        } else {
+            $this->theme = null;
+            $this->name = t('Active Theme');
+            $this->presets = [];
+        }
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @see \Concrete\Core\Area\Layout\Preset\Provider\ProviderInterface::getName()
+     */
     public function getName()
     {
-        return 'Active Theme';
+        return $this->name;
     }
 
+    /**
+     * {@inheritdoc}
+     *
+     * @see \Concrete\Core\Area\Layout\Preset\Provider\ProviderInterface::getPresets()
+     */
     public function getPresets()
     {
         return $this->presets;
     }
+
+    /**
+     * Get the theme that provides the presets.
+     */
+    public function getTheme(): ?Theme
+    {
+        return $this->theme;
+    }
+
+    /**
+     * Get the handle of the theme that provides the presets.
+     */
+    public function getThemeHandle(): string
```

Comment:
And this is needed for the changes in the CoreAreaLayout controller.

Reply:
👍